### PR TITLE
deprecate the additional architectures in prowgen config, refactor multi_arch

### PR DESCRIPTION
**Pull request by droslean**
Removes the `additional_architectures` configuration in prowgen config. Force the images presubmits to run on `build09`when the `multi_arch` is enabled.

/cc @openshift/test-platform @danilo-gemoli @deepsm007 

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by droslean** [comment]
/hold

Need a follow-up PR to the release repo. @deepsm007 Feel free to tag this if there are no suggestions. I will hold cancel and follow-up in the release repo.

File: pkg/prowgen/prowgen.go [hunk 1]
```diff
@@ -100,17 +99,25 @@ func GenerateJobs(configSpec *cioperatorapi.ReleaseBuildConfiguration, info *Pro
 			presubmitTargets = append(presubmitTargets, "[release:latest]")
 		}
 		jobBaseGen := newJobBaseBuilder().TestName("images")
+		if info.Config.MultiArch {
```
Comment (danilo-gemoli) [inline pkg/prowgen/prowgen.go:102]: You could add this logic inside the builder once and for all

File: pkg/prowgen/prowgen.go [hunk 2]
```diff
@@ -100,17 +99,25 @@ func GenerateJobs(configSpec *cioperatorapi.ReleaseBuildConfiguration, info *Pro
...
 		presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest(jobBaseGen, "images", info))
 
 		if configSpec.PromotionConfiguration != nil {
-			postsubmitsForPromotion, err := generatePostsubmitsForPromotion(newJobBaseBuilderForPromotion, info, func(options *generatePostsubmitOptions) {
-				options.imageTargets = imageTargets
-			})
-			if err != nil {
-				return nil, fmt.Errorf("error generating postsubmits for promotion: %w", err)
+			jobBaseGen = newJobBaseBuilder().TestName("images")
+			if info.Config.MultiArch {
```
Comment (danilo-gemoli) [inline pkg/prowgen/prowgen.go:110]: Same as above here

File: pkg/prowgen/prowgen.go (hunk 1 above)
Comment (droslean) [reply pkg/prowgen/prowgen.go:102]: We don't want that. We need to mutate only the images presubmits and postsubmit.

Comment (danilo-gemoli) [reply pkg/prowgen/prowgen.go:102]: Got it

**Pull request merged**